example/issue_comment.go

```diff
@@ -66,22 +82,98 @@ func (h *PRCommentHandler) Handle(ctx context.Context, eventType, deliveryID str
 	repoOwner := repo.GetOwner().GetLogin()
 	repoName := repo.GetName()
 	author := event.GetComment().GetUser().GetLogin()
+	commentID := event.GetComment().GetID()
 	body := event.GetComment().GetBody()
 
-	if strings.HasSuffix(author, "[bot]") {
-		logger.Debug().Msg("Issue comment was created by a bot")
+	if !strings.HasPrefix(author, targetAuthor) {
+		logger.Debug().Msg(fmt.Sprintf("Issue comment was not created by the user: %s", targetAuthor))
 		return nil
 	}
 
-	logger.Debug().Msgf("Echoing comment on %s/%s#%d by %s", repoOwner, repoName, prNum, author)
-	msg := fmt.Sprintf("%s\n%s said\n```\n%s\n```\n", h.preamble, author, body)
-	prComment := github.IssueComment{
-		Body: &msg,
+	// fetch the prow URL
+	r, _ := regexp.Compile(`(https:\/\/prow.ci.openshift.org\/view\/gs\/origin-ci-test\/pr-logs\/pull.*)\)`)
+	prowJobURL := r.FindStringSubmatch(body)[1]
+
+	// process the test failures from the prow URL
+	cfg := prow.ScannerConfig{
+		ProwJobURL:      prowJobURL,
+		FileNameFilter: []string{junitFilename}, // cross check its targets only the junit.xml within the ....
 	}
 
-	if _, _, err := client.Issues.CreateComment(ctx, repoOwner, repoName, prNum, &prComment); err != nil {
-		logger.Error().Err(err).Msg("Failed to comment on pull request")
+	scanner, err := prow.NewArtifactScanner(cfg)
+	if err != nil {
+		return fmt.Errorf("failed to initialize artifact scanner: %+v", err)
+	}
+
+	if err := scanner.Run(); err != nil {
+		return fmt.Errorf("failed to scan artifacts for prow job %s: %+v", prowJobURL, err)
+	}
+
+	failedTestCasesNames := getFailedTestCases(scanner, logger)
+
+	// Update the comment body with the names of failed testcases
+	if len(failedTestCasesNames) > 0 {
+		logger.Debug().Msgf("Updating comment with ID:%d %s/%s#%d by %s", commentID, repoOwner, repoName, prNum, author)
+
+		msg := "**List of E2E tests that failed in the latest CI run**: \n"
+		for _, testcaseName := range failedTestCasesNames {
+			msg = msg + fmt.Sprintf("\n* %s\n", testcaseName)
+		}
+		msg = msg + "\n-------------------------------\n\n" + body
+
+		prComment := github.IssueComment{
+			Body: &msg,
+		}
+
+		err := wait.PollUntilContextTimeout(context.Background(), 3*time.Second, 10*time.Minute, true, func(context.Context) (done bool, err error) {
+			if _, _, err := client.Issues.EditComment(ctx, repoOwner, repoName, commentID, &prComment); err != nil {
+				logger.Error().Err(err).Msg("Failed to edit comment...retrying")
+				return false, nil
+			}
+
+			return true, nil
+		})
+
+		if err != nil {
+			logger.Error().Err(err).Msg(fmt.Sprintf("Failed to edit comment, will stop processing this comment with ID: %v", commentID))
+		}
 	}
 
 	return nil
 }
+
+func getFailedTestCases(scanner *prow.ArtifactScanner, logger zerolog.Logger) []string {
```

Comment:
One missing thing is to cover the case when the given prow URL fails during the infra-setup.

Reply:
Addressed with https://github.com/dheerajodha/go-githubapp/pull/1/commits/7e350f841f026d0c0b649ddc5affaed874b9e922